mesa/mesaPROTON_OC.py

```diff
@@ -244,18 +249,27 @@ def intervention_is_on(self):
         return self.ticks % self.ticks_between_intervention == 0 and self.intervention_start <= self.ticks < self.intervention_end
 
     def socialization_intervene(self):
-        potential_targets = [x for x in schedule.agents if x.age() < 18 and x.age >= 6 and x.my_school != None]
-        targets = self.rng.choice(potential_targets,
-                                   p=[x.criminal_tendency for x in potential_targets],
-                                   size=math.ceil((targets_addressed_percent / 100 * len(potential_targets))
-                                                  # criminal_tendency + criminal_tendency_addme_for_weighted_extraction
-                                                  )
-                                   )
-        if social_support == "educational" or social_support == "all": self.soc_add_educational(targets)
-        if social_support == "psychological" or social_support == "all": self.soc_add_psychological(targets)
-        if social_support == "more friends" or social_support == "all": self.soc_add_more_friends(targets)
-        welfare_createjobs({x for x in schedule.agents if
-                            x.gender_is_male == False and x.neighbors.get('offspring').intersection(set(targets))})
+        """
+        This procedure is active when the self.social_support attribute is different from None. There are 4
+        possible social interventions: educational, psychological, more-friends or all. The intervention has effect on a
+        portion of eligible members, determined by "how_many" variable that depends on self.targets_addressed_percent attribute.
+        The interventions consist of:
+        1. soc_add_educational, the max_education_level attribute of the eligible members is increased by one
+        2. soc_add_psychological, a new support member (who has not committed crimes) is added to the friends network
+        3. soc_add_more_friends, a new support member (with a low level of tendency to crime) is added to the friends network
+        4. welfare_createjobs, new jobs are created and assigned to eligible members (mothers)
+        :return: None
+        """
+        potential_targets = [agent for agent in self.schedule.agents if
+                             agent.age() <= 18 and agent.age() >= 6 and agent.my_school != None]
+        how_many = int(np.ceil(self.targets_addressed_percent / 100 * len(potential_targets)))
```

Comment:
Why you need the int here? Ceil must be a int...

Reply:
For us it's an int. But for our python interpreter, if how_many is 10.0 it's a float and numpy.default_rng.choice() takes integers. 
In any case, it is not necessary here because inside the function extra.weighted_n_of() we already transform this parameter into int.